src/os/bluestore/BlueStore.cc

```diff
@@ -2003,6 +2005,28 @@ void BlueStore::ExtentMap::update(KeyValueDB::Transaction t,
   }
 }
 
+bid_t BlueStore::ExtentMap::allocate_spanning_blob_id()
+{
+  if(spanning_blob_map.empty())
+    return 0;
+  bid_t bid = spanning_blob_map.rbegin()->first + 1;
+  // bid is valid and available.
+  if (bid >= 0)
+    return bid;
+  // Find next unused bid;
+  bid = rand()%(numeric_limits<bid_t>::max()+1);
+  auto begin_bid = bid;
+  do {
+    if (!spanning_blob_map.count(bid))
+      return bid;
+    else {
+      bid++;
+      if(bid < 0) bid = 0;
+    }
+  } while(bid != begin_bid);
+  return -1;
```

Comment:
could `assert()` here instead of where this method is called. as we never expect a collision.